### PR TITLE
WIP

### PR DESCRIPTION
**Pull request by artek-koltun**

### CODE REVIEW

File: build/storage/tests/ut/host-target/fake_pci_sys_fs.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+                    self.bind_pci_device_to_driver(PciAddress(str(content)))
+                else:
+                    self.need_to_bind_device_to_driver = True
+            elif file == self.get_unbind_path():
+                if self.need_to_unbind_device_from_driver:
+                    self.unbind_pci_device_to_driver(PciAddress(str(content)))
+                else:
+                    self.need_to_unbind_device_from_driver = True
+            elif os.path.basename(file) == "sriov_numvfs":
+                self._sriov_vfs_set(file, str(content))
```
Comment (intelfisz) [inline build/storage/tests/ut/host-target/fake_pci_sys_fs.py:130]: make sure internal function does not get wrong input - validate before num_vfs <= initial_vfs (<= total_vfs)

File: build/storage/core/host-target/device_exerciser.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    def run_fio(
+        self, device_handle: str, volume_ids: set[VolumeId], fio_args: FioArgs
+    ) -> str:
+        storage_device = self._create_storage_device(device_handle)
+        return storage_device.run_fio_on_volumes(fio_args, volume_ids)
+
+    def _create_storage_device(self, device_handle: str) -> StoragePcieDevice:
+        sma_handle = self._create_sma_handle(device_handle)
+        logging.info("SMA Handle object is created")
+        if sma_handle.get_protocol() == VIRTIO_BLK_PROTOCOL:
```
[the remark this answers is not in the document]
Comment (artek-koltun) [reply build/storage/core/host-target/device_exerciser.py:60]: `*Device` classes do not have any state within them we need to track between calls, so it is ok to recreate them every time

File: build/storage/tests/ut/host-target/fake_pci_sys_fs.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+import os
+import glob
+import shutil
+from helpers.file_helpers import read_file, write_file
+from pci  import PciAddress
+
+# TODO create tests which are tun against this fake fs and real sysfs
```
[the remark this answers is not in the document]
Comment (artek-koltun) [reply build/storage/tests/ut/host-target/fake_pci_sys_fs.py:12]: done

File: build/storage/tests/ut/host-target/fake_pci_sys_fs.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+            vf_plugged_device_dir = self.get_bound_device_path(vf_pci_addr)
+            os.unlink(vf)
+            if os.path.exists(vf_plugged_device_dir):
+                os.unlink(vf_plugged_device_dir)
+            shutil.rmtree(self.get_pci_device_path(vf_pci_addr))
+
+    def _vf_index_to_pci_address(self, index):
+        bus = hex(index // 256 % 256 + self.vf_base_bus)[2:].zfill(2)
+        device = hex((index // 8 % 32))[2:].zfill(2)
+        function = hex(index % 8)[2:].zfill(1)
```
[the remark this answers is not in the document]
Comment (artek-koltun) [reply build/storage/tests/ut/host-target/fake_pci_sys_fs.py:50]: put a comment

**Review by intelfisz — APPROVED** [review]

**Pull request merged**